spec.bs

```diff
@@ -243,8 +250,200 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
+  1. Set |result|'s {{URLPatternResult/hostname}} to the result of [=create a component match result=] given |urlpattern|'s [=URLPattern/hostname component=], |hostname|, and |hostnameExecResult|.
+  1. Set |result|'s {{URLPatternResult/port}} to the result of [=create a component match result=] given |urlpattern|'s [=URLPattern/port component=], |port|, and |portExecResult|.
+  1. Set |result|'s {{URLPatternResult/pathname}} to the result of [=create a component match result=] given |urlpattern|'s [=URLPattern/pathname component=], |pathname|, and |pathnameExecResult|.
+  1. Set |result|'s {{URLPatternResult/search}} to the result of [=create a component match result=] given |urlpattern|'s [=URLPattern/search component=], |search|, and |searchExecResult|.
+  1. Set |result|'s {{URLPatternResult/hash}} to the result of [=create a component match result=] given |urlpattern|'s [=URLPattern/hash component=], |hash|, and |hashExecResult|.
+  1. Return |result|.
+</div>
+
+<div algorithm>
+  To <dfn>process a URLPatternInit</dfn> given a {{URLPatternInit}} |init|, a string |type|, a string |protocol|, a string |username|, a string |password|, a string |hostname|, a string |port|, a string |pathname|, a string |search|, and a string |hash|:
+
+  1. Let |result| be the result of creating a new {{URLPatternInit}}.
+  1. Set |result|'s {{URLPatternInit/protocol}} to |protocol|.
+  1. Set |result|'s {{URLPatternInit/username}} to |username|.
+  1. Set |result|'s {{URLPatternInit/password}} to |password|.
+  1. Set |result|'s {{URLPatternInit/hostname}} to |hostname|.
+  1. Set |result|'s {{URLPatternInit/port}} to |port|.
+  1. Set |result|'s {{URLPatternInit/pathname}} to |pathname|.
+  1. Set |result|'s {{URLPatternInit/search}} to |search|.
+  1. Set |result|'s {{URLPatternInit/hash}} to |hash|.
+  1. If |init|'s {{URLPatternInit/baseURL}} is not null:
+    1. Let |baseURL| be the result of [=URL parser|parsing=] |init|'s {{URLPatternInit/baseURL}}.
+    1. If |baseURL| is failure, then throw a TypeError.
+    1. Set |result|'s {{URLPatternInit/protocol}} to |baseURL|'s [=URL/protocol].
+    1. Set |result|'s {{URLPatternInit/username}} to |baseURL|'s [=URL/username].
+    1. Set |result|'s {{URLPatternInit/password}} to |baseURL|'s [=URL/password].
+    1. Set |result|'s {{URLPatternInit/hostname}} to |baseURL|'s [=URL/hostname].
+    1. Set |result|'s {{URLPatternInit/port}} to |baseURL|'s [=URL/port].
+    1. Set |result|'s {{URLPatternInit/pathname}} to |baseURL|'s [=URL/pathname].
+    1. Set |result|'s {{URLPatternInit/search}} to |baseURL|'s [=URL/search].
+    1. Set |result|'s {{URLPatternInit/hash}} to |baseURL|'s [=URL/hash].
+  1. If |init|'s {{URLPatternInit/protocol}} is not null:
+    1. Set |result|'s {{URLPatternInit/protocol}} to the result of [=canonicalize protocol=] given |init|'s {{URLPatternInit/protocol}} and |type|.
```

Comment:
IMO if you have a nested list with a single step, you should just collapse it into a single sentence.